### ISSUE TITLE
Foxx route tagging

### ISSUE THREAD

**Issue by caracal7** [arangodb/arangodb#2338]
Is it possible to organise endpoints in FOXX service API tab?

I have more then 50 endpoints in one service and it will be cool to separate endpoints by categories.
For example:

Default
GET /schemes_list 
GET /schemes_list2 
...
...
...
Schemes
DELETE /schemes/{name} 
GET /schemes/{name} 
POST /schemes/{name} 
....
...
Nodes
GET /db/list 
GET /db/node/{id} 
POST /db/node/{id} 
....
...
Other category
....
...
...

**Comment by dothebart** [comment]
Hm, doesn't sound like a Microservice anymore? ;-) 

If you have a look at the arangodb swagger documentation, these are also combined by topic, so its generally possible.

**Comment by caracal7** [comment]
I have about 10 collections and I don't think that it it too many :)
And for every collection I need CRUD - 4 operations(endpoints). And 10*4=40 endpoints and it is not a big huge service, it is Microservice I think))
I don't think that it is good idea to separate this to 10 different microservices because collections is interrelated. 

I tried to find at the arangodb swagger documentation something like topics but found only .summury(), .description() and etc. It is doesn't solve my problem.

P.S. Sorry for my bad english(

**Comment by dothebart** [comment]
n/p - not a native speaker neither.

the ArangoDB API Documentation uses the 'tags' feature to group the different topics to sub topics:

```
 "/_admin/shutdown": {
            "delete": {
                "description": "\n\nThis call initiates a clean shutdown sequence.\n\n", 
                "parameters": [], 
                "responses": {
                    "200": {
                        "description": "is returned in all cases.\n\n\n"
                    }
                }, 
                "summary": " Initiate shutdown sequence", 
                "tags": [
                    "Administration"
                ], 
                "x-examples": [], 
```
@pluma is there a way to achieve this using foxx?

**Comment by pluma** [comment]
Give the 3.2 alpha a spin. Foxx now supports the `.tag` method on endpoints. I think we sneakily added the support for this in one of the recent 3.1 releases but it's officially supported in 3.2.

**Comment by pluma** [comment]
Oops, just realised this isn't documented yet.

**Comment by caracal7** [comment]
Wow) So fast response and solution.
I go to try install 3.2.alpha2 )))

**Comment by pluma** [comment]
@caracal7 sorry for the confusion.

The feature is available in **3.1.10** and later. So you don't need to upgrade to the 3.2 alpha just to use it.

You can try out the alpha if you want but beware that alpha releases are expected to be unstable and break in various exciting ways.

**Comment by caracal7** [comment]
@pluma Thanks a lot
I have setup new server with 3.1.12. It working exactly as I need)))
